test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -580,6 +581,12 @@ private static Settings getRandomNodeSettings(long seed) {
             int retryTimeoutSeconds = RandomNumbers.randomIntBetween(random, 0, 60);
             builder.put(TransportReplicationAction.REPLICATION_RETRY_TIMEOUT.getKey(), timeValueSeconds(retryTimeoutSeconds));
         }
+        if (random.nextInt(10) == 0) {
+            builder.put(
+                PersistedClusterStateService.DOCUMENT_PAGE_SIZE.getKey(),
+                new ByteSizeValue(RandomNumbers.randomIntBetween(random, rarely() ? 10 : 100, 1000))
+            );
+        }
```

Comment:
I wonder if we should apply this more often but use between(10, 1000000) to allow for more variety of cases?

```suggestion
        if (randomBoolean()) {
            builder.put(
                PersistedClusterStateService.DOCUMENT_PAGE_SIZE.getKey(),
                new ByteSizeValue(RandomNumbers.randomIntBetween(random, rarely() ? 10 : 100, 1000000))
            );
        }
```

Reply:
I think the more interesting corner cases are mostly when paginating harder; simply increasing the limit to 1000000 means that these interesting cases get much much less likely, but I introduced a variety of limits in ade43436fd8.